fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1558,6 +1560,42 @@ public Collection<OverdueLoanScheduleData> retrieveAllLoansWithOverdueInstallmen
         return this.jdbcTemplate.query(sqlBuilder.toString(), rm, penaltyWaitPeriod, penaltyWaitPeriod);
     }
 
+    @Override
+    public Collection<OverdueLoanScheduleData> retrieveAllOverdueInstallmentsForLoan(final Loan loan) {
+        Collection<OverdueLoanScheduleData> list = new ArrayList<>();
+
+        if (!loan.isOpen()) {
+            return list;
+        }
+        final Long penaltyWaitPeriod = configurationDomainService.retrievePenaltyWaitPeriod();
+        final boolean backdatePenalties = configurationDomainService.isBackdatePenaltiesEnabled();
+
+        for (LoanRepaymentScheduleInstallment installment : loan.getRepaymentScheduleInstallments()) {
+            if (installment.isObligationsMet() || installment.isRecalculatedInterestComponent()) {
+                continue;
+            }
+
+            if (installment.isOverdueOn(DateUtils.getBusinessLocalDate().plusDays(penaltyWaitPeriod))) {
+                if (!backdatePenalties
+                        && installment.getDueDate().compareTo(DateUtils.getBusinessLocalDate().plusDays(penaltyWaitPeriod + 1)) < 0) {
+                    continue;
+                }
+                Optional<Charge> penaltyCharge = loan.getLoanProduct().getLoanProductCharges().stream()
```

Comment:
Is there a way we could extract this magic number (the 9) into a constant or reuse it from somewhere?